packages/expo-enclave/android/src/main/java/expo/modules/enclave/BiometricsKeyManager.kt

```diff
@@ -0,0 +1,126 @@
+    val activityProvider: ActivityProvider = moduleRegistry.getModule(ActivityProvider::class.java)
+    return activityProvider.currentActivity
+  }
+
+  internal fun createSigningPrivkey(accountName: String) {
+    var params = KeyGenParameterSpec.Builder(accountName, KeyProperties.PURPOSE_SIGN or KeyProperties.PURPOSE_VERIFY)
+      .setAlgorithmParameterSpec(ECGenParameterSpec("secp256r1"))
+      .setDigests(KeyProperties.DIGEST_SHA256)
+      .setUserAuthenticationRequired(true)
+      .setUserAuthenticationParameters(0, KeyProperties.AUTH_BIOMETRIC_STRONG or KeyProperties.AUTH_DEVICE_CREDENTIAL)
+      .setInvalidatedByBiometricEnrollment(false)
+
+    if (useStrongbox) {
+      params.setIsStrongBoxBacked(true)
+    }
+
+    var generator = params.build()
+    var keyPairGenerator = KeyPairGenerator.getInstance(KeyProperties.KEY_ALGORITHM_EC, KEYSTORE_PROVIDER)
+    keyPairGenerator.initialize(generator)
+    keyPairGenerator.generateKeyPair()
+  }
+
+  internal fun getSigningPrivkey(accountName: String): KeyPair {
+    val ks: KeyStore = KeyStore.getInstance(KEYSTORE_PROVIDER).apply { load(null) }
+    if (!ks.containsAlias(accountName)) {
+      throw KeyStoreException("Unable to read account private key")
+    }
+    val entry = ks.getEntry(accountName, null)
+    if (entry !is KeyStore.PrivateKeyEntry) {
+        throw KeyStoreException("Bad format for account private key")
+    }
+    return KeyPair(entry.certificate.publicKey, entry.privateKey)
+  }
+
+  override fun fetchPublicKey(accountName: String): String? {
+    return getSigningPrivkey(accountName).public.encoded.toHexString()
+  }
+  
+  override fun createKeyPair(accountName: String): String {
+    createSigningPrivkey(accountName)
+    return fetchPublicKey(accountName)!!
+  }
+
+  override fun sign(accountName: String, hexMessage: String, promise: Promise) {
+    val fragmentActivity = getCurrentActivity() as FragmentActivity?
+    val biometricPrompt = BiometricPrompt(
+      fragmentActivity!!, ContextCompat.getMainExecutor(context),
+      object : BiometricPrompt.AuthenticationCallback() {
+        override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+          super.onAuthenticationError(errorCode, errString)
+          promise.reject("ERR_BIOMETRIC_AUTHENTICATION_ERRORED", "Biometric authentication errored")
+        }
+
+        override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+          super.onAuthenticationSucceeded(result)
+          completeSignature(result.cryptoObject!!.getSignature(), hexMessage, promise)
+        }
+
+        override fun onAuthenticationFailed() {
+          super.onAuthenticationFailed()
+          promise.reject("ERR_BIOMETRIC_AUTHENTICATION_FAILED", "Biometric authentication failed")
+        }
+    })
+
+    val promptInfo = BiometricPrompt.PromptInfo.Builder()
+      .setTitle("Authorise transaction")
+      .setSubtitle("Sign Daimo tx " + hexMessage)
```

Comment:
we'll have to change this title and subtitle, but can do that later

i recommend making `title` and `subtitle` arguments to this function

Reply:
Good point, added a param to pass these in to the function instead :)